.pre-commit-config.yaml

```diff
@@ -0,0 +1,24 @@
+# All available hooks: https://pre-commit.com/hooks.html
+# R specific hooks: https://github.com/lorenzwalthert/precommit
+repos:
+-   repo: https://github.com/lorenzwalthert/precommit
+    rev: v0.4.2
+    hooks:
+    -   id: style-files
```

Comment:
This one can be quite annoying, to be honest. It often makes changes unrelated to the PR.

Reply:
The way I understand stylers/linters like this is that making changes unrelated to the PR only happens if the style is not upheld to begin with. This is a common thing with for example `prettier` in JavaScript too. Is the annoyance with the styler or with the style itself?

This will not happen so much with new instances of **packagetemplate** inasmuch for existing packages. For new instances, the pre-commit hook ensures the style is applied from the beginning, hence, it will only make changes where the PR is doing things anyway. In older packages, if style is not consistent, it would cause changes upon first run to get everything in style. We could opt to run the styler before setting up the pre-commit hook to ensure PRs don't get clogged unnecessarily - which is a valid point by @jamesmbaazam 👍

Reply:
> We could opt to run the styler before setting up the pre-commit hook to ensure PRs don't get clogged unnecessarily - which is a valid point by @jamesmbaazam 👍

Yes, that's a good workaround. In the past, I've not included the style changes to a PR if it's not within its scope as it can be distracting to review.